projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_plugin.py

```diff
@@ -53,6 +65,9 @@ def impala_query(ctx: click.Context, query):
 
 
 class ImpalaPlugin:
+    def __init__(self, lineage_logger=None):
```

Comment:
My preference would be to be a bit more encapsulated and have "LineagePlugin" class which contains all the lineage related hooks . Better separation of responsibilities and decoupled code. 
Otherwise this file is becoming dumpster for everything. 

The only change then in this file would be in vdk_start hook to register the new Lineage Plugin 

```
    plugin_registry.load_plugin_with_hooks_impl(LineagePlugin())
```

Reply:
Umm, I do not agree on 100% here as, but I do not disagree also 😄 . Currently the linage logic is encapsulated in separate module - lineage_utils and only the hook is defined in impala_plugin. This way you can see all impala related hooks in a single place which makes it easier to understand what is happening.
Anyhow I will perform the suggested refactoring mainly because I dislike how complex the impala_plugin module has become. For example vdk-trino seems much more streamlined.